File: Programing_Basics_with_JS/conditionalStatementsMoreExercises/03.Harvest.js
function harvest(input) {
    let vineyard = Number(input[0]);
    let grapesPerMeter = Number(input[1]);
    let neededWineLiters = Number(input[2]);
    let numberOfWorkers = Number(input[3]);

    let wineVineyard = vineyard * 0.4;
    let allGrapes = grapesPerMeter * wineVineyard;
    let litersWine = allGrapes / 2.5;

    let wineDiff = Math.abs(neededWineLiters - litersWine);
    let wineForOneWorker = wineDiff / numberOfWorkers;

    if (neededWineLiters > litersWine) {
        console.log(`It will be a tough winter! More ${Math.floor(wineDiff)} liters wine needed.`);
    } else {
        console.log(`Good harvest this year! Total wine: ${Math.floor(litersWine)} liters.`);
        console.log(`${Math.ceil(wineDiff)} liters left -> ${Math.ceil(wineForOneWorker)} liters per person.`);
    }
}

harvest(['650', '2', '175', '3'])
harvest(['1020', '1.5', '425', '4'])

File: Programming_Basics_with_Python/conditional_statements_more_exercise/03._harvest.py
from math import floor, ceil


def harvest(vineyard, grapes_from_one_meter, needed_wine, workers, vineyard_percent_for_wine, grape_for_one_litter_wine):
    all_grape = vineyard * grapes_from_one_meter
    grape_for_wine = all_grape * vineyard_percent_for_wine
    produced_wine = grape_for_wine / grape_for_one_litter_wine
    wine_difference = produced_wine - needed_wine

    if produced_wine >= needed_wine:
        wine_per_person = wine_difference / workers
        return f'Good harvest this year! Total wine: {floor(produced_wine)} liters.\n{ceil(wine_difference)} liters left -> {ceil(wine_per_person)} liters per person.'
    else:
        return f'It will be a tough winter! More {(floor(abs(wine_difference)))} liters wine needed.'


vineyard = int(input())
grapes_from_one_meter = float(input())
needed_wine = int(input())
workers = int(input())

vineyard_percent_for_wine = 0.4
grape_for_one_litter_wine = 2.5

print(harvest(vineyard, grapes_from_one_meter, needed_wine, workers, vineyard_percent_for_wine, grape_for_one_litter_wine))